contents/ruby/Rails.md

```diff
@@ -0,0 +1,53 @@
+<frontmatter>
+  title: Introduction to Ruby on Rails
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Ruby on Rails
+
+Authors: [Chattoraj Ayush](https://github.com/AyushChatto)
+
+## Ruby on Rails overview
+
+**Ruby on Rails** is an **easy-to-learn** framework that allows you to develop powerful web applications **quickly** and with as little effort as possible. It was developed in the early 2000s by Danish Programmer David Heinemeier Hansson, or 'DHH'. 
+
+Ruby on Rails, or simply Rails, gained widespread popularity in the 2000s, and continues to be one of the most popular web-frameworks even to this day, because of how quickly it allows an easily scalable minimum viable product to be built, making it the perfect choice for startups. This is owed in large part to the Rails Doctrines of **Optimising for Programmer Happiness** and **Convention over Configuration**.
+
+> ... I created Rails for me. To make me smile, first and foremost. Its utility was to many degrees subservient to its ability to make me enjoy my life more. To enrich my daily toil of wrangling requirements and requests for web information systems.
+>
+> ... One of the early productivity mottos of Rails went: “You’re not a beautiful and unique snowflake”. It postulated that by giving up vain individuality, you can leapfrog the toils of mundane decisions, and make faster progress in areas that really matter.
+>
+> *David Heinemeier Hansson,* __The Rails Doctrine__
+
+Many popular and very technically mature websites continue to use Rails. These include: GitHub, Airbnb, Twitter, Hulu, Shopify, and Twitch, among others.
+
+## Getting Started
+
+The official website for [Rails](https://rubyonrails.org) will have all the information needed to initialize your first Rails application.
+
+To install Rails on your computer, you will first need to install [Ruby](https://www.ruby-lang.org/en/). After this, you simply need to run:
+
+```rb
+gem install rails
+```
+gem is a package manager for Ruby that allows you to install Rails. Once you're done with the setup, you can start developing applications right away. While the [Rails Guides](https://guides.rubyonrails.org/) are the best possible place to learn everything about Rails, [Go Rails](https://gorails.com/) also has great screencast tutorials that really hold your hand through the multitude of features Rails has, and can be used by more visual learners.
+
+We will now explore some of the unique features of Rails that make it such a powerful tool.
+
+### Optimised for Programmer Happiness
```

Comment:
As this is more of a Ruby concept than a Rails concept (although they are of course very intertwined), maybe mention how so much of Rails is really about its Ruby foundation.

Reply:
Actually, the Ruby concept is Principle of Least Surprise - The Principle of the Bigger Smile is primarily a Rails concept :) 
https://rubyonrails.org/doctrine/#optimize-for-programmer-happiness

Reply:
Aha, I guess this just goes to show how there's not too much value trying to distinguish between strictly Rails vs strictly Ruby 😛